code/game/machinery/computer/robot.dm

```diff
@@ -20,221 +20,179 @@
-// Parameters: 1 (operator - mob which is operating the console.)
-// Description: Returns NanoUI-friendly list of accessible cyborgs.
-/obj/machinery/computer/robotics/proc/get_cyborgs(var/mob/operator)
-	var/list/robots = list()
-
-	for(var/mob/living/silicon/robot/R in GLOB.mob_list)
-		// Ignore drones
-		if(istype(R, /mob/living/silicon/robot/drone))
-			continue
-		// Ignore antagonistic cyborgs
-		if(R.scrambledcodes)
-			continue
-
-		var/list/robot = list()
-		robot["name"] = R.name
-		if(R.stat)
-			robot["status"] = "Not Responding"
-		else if(!R.canmove)
-			robot["status"] = "Lockdown"
-		else
-			robot["status"] = "Operational"
-
-		if(R.cell)
-			robot["cell"] = 1
-			robot["cell_capacity"] = R.cell.maxcharge
-			robot["cell_current"] = R.cell.charge
-			robot["cell_percentage"] = round(R.cell.percent())
-		else
-			robot["cell"] = 0
-
-		var/turf/pos = get_turf(R)
-		var/area/bot_area = get_area(R)
-		robot["xpos"] = pos.x
-		robot["ypos"] = pos.y
-		robot["zpos"] = pos.z
-		robot["area"] = format_text(bot_area.name)
-
-		robot["health"] = round(R.health * 100 / R.maxHealth,0.1)
-
-		robot["module"] = R.module ? R.module.name : "None"
-		robot["master_ai"] = R.connected_ai ? R.connected_ai.name : "None"
-		robot["hackable"] = 0
-		// Antag AIs know whether linked cyborgs are hacked or not.
-		if(operator && istype(operator, /mob/living/silicon/ai) && (R.connected_ai == operator) && (operator.mind.special_role && operator.mind.original == operator))
-			robot["hacked"] = R.emagged ? 1 : 0
-			robot["hackable"] = R.emagged? 0 : 1
-		robots.Add(list(robot))
-	return robots
-
-// Proc: get_cyborg_by_name()
-// Parameters: 1 (name - Cyborg we are trying to find)
-// Description: Helper proc for finding cyborg by name
-/obj/machinery/computer/robotics/proc/get_cyborg_by_name(var/name)
-	if(!name)
-		return
-	for(var/mob/living/silicon/robot/R in GLOB.mob_list)
-		if(R.name == name)
-			return R
+			R.self_destruct()
+			. = TRUE
+		if("stopbot") // lock or unlock the borg
+			if(isrobot(usr))
+				to_chat(usr, "<span class='danger'>Access Denied.</span>")
+				return
+			var/mob/living/silicon/robot/R = locateUID(params["uid"])
+			if(!can_control(usr, R, TRUE))
+				return
+			message_admins("<span class='notice'>[ADMIN_LOOKUPFLW(usr)] [!R.lockcharge ? "locked down" : "released"] [ADMIN_LOOKUPFLW(R)]!</span>")
+			log_game("[key_name(usr)] [!R.lockcharge ? "locked down" : "released"] [key_name(R)]!")
+			R.SetLockdown(!R.lockcharge)
+			to_chat(R, "[!R.lockcharge ? "<span class='notice'>Your lockdown has been lifted!" : "<span class='alert'>You have been locked down!"]</span>")
+			if(R.connected_ai)
+				to_chat(R.connected_ai, "[!R.lockcharge ? "<span class='notice'>NOTICE - Cyborg lockdown lifted</span>" : "<span class='alert'>ALERT - Cyborg lockdown detected</span>"]: <a href='?src=[R.connected_ai.UID()];track=[html_encode(R.name)]'>[R.name]</a></span><br>")
+			. = TRUE
+		if("magbot") // AIs hacking/emagging a borg
```

Comment:
`magbot`? Is this a typo from `emagbot`?

Reply:
TG calls it "magbot", I assume as an abbreviation.
I'm just renaming it to "hackbot" as that is clearer for what it actually does.